src/kademlia/traversal_algorithm.cpp

```diff
@@ -593,6 +593,28 @@ void traversal_algorithm::status(dht_lookup& l)
 	l.last_sent = last_sent;
 }
 
+void traversal_algorithm::look_for_nodes(bdecode_node const& r,bool const is_traverse)
```

Comment:
I don't think a raw ``bool`` is a good idea. it's impossible to tell what it means at the call sites. Also, it seems like this function is special-tailored to the two places it's called from and not necessarily a reasonable API boundary.
It would probably be reasonable to take a ``std::function<void(node_endpoint)>`` instead of a bool, and let the call sites decide exactly what to do with the nodes. then you could just feed the node_endpoint straight into the callback function, and use a one-line lambda at the call sites